### PR TITLE
Raftstore: fix split skip bug due to incorrectly setting has_calculated_region_size

### LINKED ISSUE #12510: Raftstore: tikv does not split regions whose  size is larger than the max region size after restart with config change
### What version of TiKV are you using?
6.0-alpha

### What operating system and CPU are you using?
CentOS 7/8

### Steps to reproduce
Set region max size/ region split size with 1GB and upload about 30GB data.  Waiting for the region count stable and then change the max size/region split size with 200MB, meanwhile enable region bucket and set region bucket size as 100MB. 
Restart tikvs.

### What did you expect?
TiKV'e regions will be splitted after the reboot.

### What did happened?
TiKV's regions will not be splitted. 

related bug: https://github.com/tikv/pd/issues/4926

### PR DESCRIPTION
**Pull request by tonyxuqqi**
Signed-off-by: qi.xu <tonxuqi@outlook.com>

### What is changed and how it works?

Issue Number: Close #12510

What's Changed:
Related bug:  tikv/pd#4926

```
rename has_calculated_region_size to may_skip_split_check and only set it  true after split check task is scheduled. 
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test
- No code
- Unit test

Side effects

None

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore/src/store/worker/split_check.rs [hunk 1]
```diff
@@ -515,6 +515,8 @@ where
                             "region_id" => region.get_id(),
                         );
                     }
+                    let region_id = region.get_id();
```
Comment (BusyJay) [inline components/raftstore/src/store/worker/split_check.rs:518]: Why not just also trigger split if necessary here?

**Comment by nolouch** [comment]
Does https://github.com/tikv/pd/issues/4926 relative to this fix?

File: components/raftstore/src/store/fsm/peer.rs [hunk 2]
```diff
@@ -1149,7 +1149,6 @@ where
     fn on_clear_region_size(&mut self) {
         self.fsm.peer.approximate_size = None;
         self.fsm.peer.approximate_keys = None;
-        self.fsm.peer.has_calculated_region_size = false;
```
Comment (BusyJay) [inline components/raftstore/src/store/fsm/peer.rs:1152]: Why remove this?

**Review by BusyJay — COMMENTED** [review]
Any test case?

**Comment by BusyJay** [comment]
> Does [tikv/pd#4926](https://github.com/tikv/pd/issues/4926) relative to this fix?

Yes.

File: components/raftstore/src/store/fsm/peer.rs (hunk 2 above)
Comment (tonyxuqqi) [reply components/raftstore/src/store/fsm/peer.rs:1152]: It's moved into register tick.

**Comment by tonyxuqqi** [comment]
> Any test case?

Should it be covered by existing test cases at least?

File: tests/integrations/raftstore/test_split_region.rs [hunk 3]
```diff
@@ -610,6 +610,45 @@ fn test_node_split_region_diff_check() {
     test_split_region_diff_check(&mut cluster);
 }
 
+// Test steps
+// set max region size/split size 2000 and put data till 100
```
Comment (BusyJay) [inline tests/integrations/raftstore/test_split_region.rs:614]: You mean `1000`?

**Review by BusyJay — APPROVED** [review]

**Comment by BusyJay** [comment]
I think the issue number is wrong.

**Comment by tonyxuqqi** [comment]
> I think the issue number is wrong.

You're right. It's PD's issue#. I just created one for TiKV.

**Comment by tonyxuqqi** [comment]
> I think the issue number is wrong.

You're right it's PD#, it should be marked as tikv/pd# instead.

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**